cinema/views.py

```diff
@@ -34,6 +36,23 @@
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        actor_names = self.request.query_params.get("actor")
+        genre_names = self.request.query_params.get("genre")
+        title = self.request.query_params.get("title")
+        if actor_names:
+            actor_list = actor_names.split(",")
+            queryset = queryset.filter(actors__last_name__in=actor_list)
+        if genre_names:
+            genre_list = genre_names.split(",")
+            queryset = queryset.filter(genres__in=genre_list)
```

Comment:
The filtering logic for genres is incorrect. It should filter by genre names, not by IDs. Consider using `genres__name__in=genre_list`.